### PR TITLE
CORE-12255 Crypto API refinements 

### PR DESCRIPTION
**Pull request by kyriathar**
- aligns with `ByteArrays` moving packages
- aligns with `KeyUtils` functions and `CompositeKey.isFulfilledBy` changed to take `Set<PublicKey>` instead of `Iterable<PublicKey>`

api PR: [#1032](https://github.com/corda/corda-api/pull/1032)

### CODE REVIEW

**Review by driessamyn — COMMENTED** [review]
LGTM

File: libs/crypto/crypto-impl/src/main/kotlin/net/corda/crypto/impl/CompositeKeyImpl.kt [hunk 1]
```diff
@@ -185,7 +185,7 @@ class CompositeKeyImpl(val threshold: Int, childrenUnsorted: List<CompositeKeyNo
      * key tree in question, and the total combined weight of all children is calculated for every intermediary node.
      * If all thresholds are satisfied, the composite key requirement is considered to be met.
      */
-    override fun isFulfilledBy(keysToCheck: Iterable<PublicKey>): Boolean {
+    override fun isFulfilledBy(keysToCheck: Set<PublicKey>): Boolean {
```
Comment (mnesbit) [inline libs/crypto/crypto-impl/src/main/kotlin/net/corda/crypto/impl/CompositeKeyImpl.kt:188]: checkFulfilledBy needs changing to use Set too as that is where in call is made

Comment (kyriathar) [reply libs/crypto/crypto-impl/src/main/kotlin/net/corda/crypto/impl/CompositeKeyImpl.kt:188]: Done

**Review by mnesbit — DISMISSED** [review]
OK

**Review by mnesbit — APPROVED** [review]
OK

**Review by driessamyn — COMMENTED** [review]
LGTM

**Review by yift-r3 — APPROVED** [review]

**Review by driessamyn — APPROVED** [review]

**Pull request merged**